### ISSUE TITLE
Threat events not always getting updated on changing data Flow properties

### ISSUE THREAD

**Issue by rsrinivasanhome** [simonec73/threatsmanager#24]
**Describe the bug**
I have a rule which is based on properties of a dataflow .  When the threat event is created it never gets deleted despite of changing the property (to prevent the rule from firing ) and clicking the Generate threats and events & mitigation button.  The threat event is removed only when I click Remove all threat events button and regenerate the threats (By Clicking Generate Threats events & Mitigations) 

**To Reproduce**
1. My flow has both source authentication and designation authentication set to 'Yes .  (It was 'No' initially when I clicked generate threat events  the first time)
![image](https://user-images.githubusercontent.com/31912418/148373551-ce02c1be-e214-4411-9936-2af1ea57d767.png)

2. My rule has set validating when authentication set to 'No'
![image](https://user-images.githubusercontent.com/31912418/148373747-7a524618-6f61-44fb-9678-16ec15ca527d.png)
when I click on test no objects are selected.  - Which is working
3. Old threat event is not removed in spite of regenerating threats by clicking on "Generate threats and mitigations". I clicked refresh also
![image](https://user-images.githubusercontent.com/31912418/148375082-db80bcb8-a523-4803-b8b0-e8e26b7aeff2.png)

**Expected behavior**
The threat event is to be removed .  Else guide me on alternative to refresh the threat events without loosing the mitigations

**Screenshots**

**Additional context**
The refreshing of threat events happens when new threat events are added (when changing the properties of the object)  but not when old threat events are removed

**Comment by simonec73** [comment]
This is by design.
In fact, the Threats are not generated dynamically, like in TMT. They are generated when you press the **Generate Threat Events & Mitigations** button in the **Analyze** ribbon, or when you open the context menu over an object and then execute **Apply Auto Gen Rules**.
So yes, the fastest way to ensure that the change is applied is to remove the assigned Threat Events by using the **Remove All Threat Events** button in the **Analyze** ribbon, and then regenerate the Threats and Mitigations.

I understand that what you are seeing may seem an inconvenience, but please consider that what you are doing does not correspond to a typical use. In fact, users would not tamper with the automatic generation rules, ever. They would work in **Simplified** mode, which does not provide that capability. And even if they work in **Expert** mode, this would not be a problem because the only users who have a need to change the rules are those who create the Templates, and they would do what you are doing to perform experiments, therefore it would be ok to regenerate the Threats & Mitigation from scratch, or even desirable to avoid incurring in problems deriving from previous dirty situations.

Bottom line, I do not see a need to change the current behavior. If you see it, please let me understand your usage scenario.

**Comment by rsrinivasanhome** [comment]
In this scenario my threat event is not triggered by a change in the auto generation rule.  It is caused by changing the properties of an object
This is the usage scenario


1. I create the threat model diagram in threats manger studio but did not set all the properties of the object correctly
2. I see a threat event generated which should not have .  
3.  I noticed the object properties have not been set correctly and now make the needed corrections and want the threat events to be updated
4. Now if click on Remove All Threat Events button in the Analyze ribbon, and then regenerate the Threats and Mitigations I will lose the mitigations assigned to other threat events.

On another note 
 If I change the property of the object to trigger a threat event it is added to the threat events on click of "Generate Threat Events & mitigation.

**Comment by simonec73** [comment]
Something you could easily do is to remove all instances of the affected Threat Type, using the Threat Event List. Then you can generate again the Threat Events on the whole Threat Model. The existing Threat Events would not be affected.

**Comment by rsrinivasanhome** [comment]
This looks like possible work around.  In the future if my usage scenario is supported it would be great

**Comment by rsrinivasanhome** [comment]
Considering the nature of the issue and the many possible times it may occur apart of the work around can a fix be considered in the backlog
This is the scenario

- I create a threat model
- I forget to change all properties of the flow
- I click Generate threats and mitigation
- I notice the wrongly generated threat and update the property of the flow 
- I click on Generate threats and mitigation again
- I still see the wrongly generated threat
